docs/architecture/provisioning-custom-configuration.md

```diff
@@ -0,0 +1,84 @@
+# Custom configuration of provisioned Runtimes
+
+## Introduction
+
+There is a need for configuring Kyma Runtimes provisioned by the Runtime Provisioner.
+For now only option is to configure Runtime manually after provisioning is finished and Kubeconfig is fetched.
+
+
+## Solution
+
+The proposed solution is to provide ability to configure the Installer overrides as a part of provisioning API.
+
+Changes to the GraphQL schema:
+```graphql
+scalar Overrides # map[string]string
+
+type KymaConfig {
+    version: String
+    modules: [KymaModule]
+    installationOverrides: InstallationOverrides
+}
+
+input KymaConfigInput {
+    version: String!
+    modules: [KymaModule!]
```

Comment:
in Kyma there is no such thing as modules

Reply:
what is the relation between module and component?

Reply:
It is not yet decided as Kyma installation is not yet implemented. I think we will go with module being effectively a single Kyma component.

Reply:
then it should be a component, not module because now it looks strange. You define an array of modules that will be installed but they are under mutation called `config` and overrides are passed as an array of all the overrides for all the components, no clear mapping between module (component) and component.

Doing API almost 1to1 with the current installer is not good as we know the current installer is not perfect and needs another version where there is a clear separation between each component per CR and not that all components are listed in one Installation CR. Take a look on this issue if you are interested with more details https://github.com/kyma-project/kyma/issues/5849

Reply:
The terms are not well defined in kyma itself. Just reading the folder names: A component in kyma results in a docker image (see "components" folder) and gets used in a "resources" aka helm chart.

I tried to introduce the term "module" for naming the helm charts, so that kyma consists of a set of modules where a module references either 3party things or kyma components, see also https://github.com/kyma-project/kyma/blob/master/components/README.md

The kyma installer references components by the way..

Anyway, I fully agree that the proposal should be not a direct mapping of the installer :)

Reply:
So I guess we might have a little different terminology about `module` and `component`.
After talking with @a-thaler we clarified that by module we mean one Helm chart or entry in Installation CR for now. So in the API we are talking about modules, not components.

Reply:
but we cannot just like that in a side api design decide that now we have some idea of `module` in Kyma. In Kyma there are only components, this is why Installer CR have a list of components, and documentation in navigation lists components, not `modules`. If we want to change the glossary, this is not the place for it

Reply:
So if the `component` is an official name for now, we go with `component` instead of `module`.

Reply:
yep, we should not fix the glossary here, I'm fine.